tasks/fieldStatusTracker.go

```diff
@@ -0,0 +1,137 @@
+package tasks
+
+import (
+	"errors"
+	"reflect"
+)
+
+type FieldStatus struct {
+	Name          string
+	HasNewValue   bool
+	ChangeApplied bool
+	Clear         bool
+}
+
+type fieldStatusMap map[string]FieldStatus
+
+type FieldStatusTracker struct {
+	fieldStatusMap fieldStatusMap
+}
+
+const (
+	TacoStructTag = "taco"
+)
+
+var (
+	ErrFieldNotFoundInTracker = errors.New("field key not found in tracker")
+)
+
+func newFieldStatusTracker() (tr *FieldStatusTracker) {
+	return &FieldStatusTracker{
+		fieldStatusMap: map[string]FieldStatus{},
+	}
+}
+
+func (tr *FieldStatusTracker) BuildFieldMap(t Task) {
```

Comment:
It seems this struct has 2 different things it's doing. Tracking actual field status and building this map with reflection. I think it would be nicer to have this separated out.

Reply:
yes, i realised that about 2/3rd the way in. had one go at refactoring but was running out of time so decided to leave it. given that you've raised it too, i've had another go, so now there are two separate interfaces. one for name mapping and one of field status tracking. i think much better and separates the concerns fairly cleanly. note that all the tasks need the `mapper` but only the realvnc server task needs the `tracker`. when i push the latest code, please take a look and feel free to suggest improvements.